# Introduction To Probability
## Challenge 1

A and B are events of a probability space with $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$

Which of the following statements are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$

In [ ]:
"""
Answer can be found in the Markdown cell below
"""

* **Statement 1**: $P(A \cup B) = 0.6$
  
$P(A\cup B)=P(A)+P(B)-P(A\cap B)$

Thus

$P(A\cup B)$ 
= 0.3 + 0.6 - 0.1 = 0.8

The first statement is false

* **Statement 2**: $P(A \cap B^{C}) = 0.2$
  
  $P(A\cap \bar{B})=P(A)-P(A\cap B)$

  Thus

$P(A\cap \bar{B})$ = 0.3 - 0.1 = 0.2 

The second statement is true

* **Statement 3**: $P(A \cap (B \cup B^{C})) = 0.4$

$(B \cup B^{C})$ is equal to 1

Thus the whole statement is equal to P(A) which is equal to 0.3 

The third statement is false

* **Statement 4**: $P(A^{C} \cap B^{C}) = 0.3$

$P(\bar{A}\cap \bar{B}) = 1 - P(A \cup B)$

Thus $P(\bar{A}\cap \bar{B})$ = 1 - 0.8 = 0.2

Statement 4 is False 

* **Statement5**:  $P((A \cap B)^{C}) = 0.9$
  
P(A and B) = 0.1 Thus $P((A \cap B)^{C})$ = 1 - P(A and B) = 1 - 0.1 = 0.9 

Statement 5 is True 



## Challenge 2
There is a box with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball out.
* Taking a white ball out after taking a black ball out.
* Taking a red ball out after taking a black and a red ball out.
* Taking a red ball out after taking a black and a red ball out with reposition.

**Hint**: Reposition means putting back the ball into the box after taking it out.

In [ ]:
"""
The results are presented below
"""

### 1. Taking a white ball out

I need to create sampel space and then probability:

Sample space = 10 + 12 + 8 = 30

P(w)= 10/30 * 100% = 1/3 * 100% = 33.3%

### 2. Taking a white ball out after taking a black ball out.

Probability of taking balck ball first: 

P(b)= 8/30

and after we want to take white ball, so probablity of that event is:

P(w)= 10/29

Thus I need to multiply these two probabilities: 

P(w|b)= 8/30 * 10/29 = 

In [1]:
8/30 * 10/29

0.09195402298850575

### 3. Taking a red ball out after taking a black and a red ball out.

Probability of taking red ball out first: 

P(r) = 12/30

After that probability of taking a black ball: 

P(b2) = 8/29

And finally probaility of taking next red ball: 

P(r3) = 11/28 (because one red ball is already out and 2 balls were taken)

The total probability will be obtained by multiplication by these 3 probabilities (result in a cell below)

In [1]:
12/30 * 8/29 * 11/28

0.043349753694581286

The probability of taking a red ball after taking balck and red is around 4%

### 4. Taking a red ball out after taking a black and a red ball out with reposition.

Probability of taking red ball out first: 

P(r) = 12/30

After that probability of taking a black ball: 

P(b2) = 8/30

And finally probaility of taking next red ball: 

P(r3) = 12/30 (because one red ball is already out and 2 balls were taken)

The total probability will be obtained by multiplication by these 3 probabilities (result in a cell below)

In [3]:
12/30 * 8/30 * 12/30

0.042666666666666665

This probability is also around 4%

## Challenge 3

You are planning to go on a picnic today but the morning is cloudy. You hate rain so you don't know whether to go out or stay home! To help you make a decision, you gather the following data about rainy days:

* 50% of all rainy days start off cloudy!
* Cloudy mornings are common. About 40% of days start cloudy. 
* This month is usually dry so only 3 of 30 days (10%) tend to be rainy. 

What is the chance of rain during the day?

In [ ]:
"""
I can use Bayes probability as I know that: 
P(R) = 0.1 - probability of the rain
P(C) = 0.4 - probability of cloudy day 
P(C|R) = 0.5 - probability of cloudy day if rain 

and I want to calculate P(R|C) so the probabilty of rain if the day started cloudly. Following the formula below I can calculate the probability of that:
"""

$P(R|C)=\frac{P(R)\ast P(C|R)}{P(C)}$

In [5]:
0.1 * 0.5/ 0.4


0.125

The probability of rain during the day is equal to 12.5%

## Challenge 4

One thousand people were asked through a telephone survey whether they thought more street lighting is needed at night or not.

Out of the 480 men that answered the survey, 324 said yes and 156 said no. On the other hand, out of the 520 women that answered, 351 said yes and 169 said no. 

We wonder if men and women have a different opinions about the street lighting matter. Is gender relevant or irrelevant to the question?

Consider the following events:
- The answer is yes, so the person that answered thinks that more street lighting is needed.
- The person who answered is a man.

We want to know if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

**Hint**: To clearly compare the answers by gender, it is best to place the data in a table.

In [37]:
# your code here
import pandas as pd
data = {'men': [324,156], 'women': [351,169]}
survey = pd.DataFrame(data)
survey['Total']=survey['men']+survey['women']
survey


men  women  Total
0  324    351    675
1  156    169    325

In [ ]:
"""
Probability of saying yes 
P(Y)=675/1000=0.675

Probability of saying no:
P(N)=325/1000=0.325

Probability of being a men:
P(M)=480/100=0.48

Probability of being a women:
P(W)=520/1000=0.52

Probaiblity of being a men and saying yes 
P(M and Y)=324/1000=0.324

I want to find probaility of saying yes if you are a men:
P(Y|M)=P(Y and M)/ P(M)

This is calculated in the cell below:
"""

In [39]:
0.324/0.48

0.675

In [ ]:
'''
Answer: Probability of saying yes if being a men is equal to 67.5%
'''

I want to check now if it is the same for women, thus: 

P(Y|W)= P(W and Y)/ P(W)

P(W and Y)= 351/1000=0.351

P(Y|W)= 0.351/0.52 = 0.675

In [40]:
0.351/0.52

0.6749999999999999

In [ ]:
'''
Final conclusion: The probability of saying yes is equal to 67.5% and it is independent form being a women or being a men.
'''